src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InternalCpxSvDiscoverFromLongReadsSpark.java

```diff
@@ -0,0 +1,228 @@
+                        ctx.broadcast(getReference()), discoverStageArgs.fastaReference, getAuthenticatedGCSOptions(),
+                        localLogger);
+
+//        new ForSimpleStrandSwitch()
+//                .inferSvAndWriteVCF(splitUpLongReads.get(RawTypes.Inv), outputDir+"/"+RawTypes.Inv.name()+".vcf",
+//                        ctx.broadcast(getReference()), discoverStageArgs.fastaReference, getAuthenticatedGCSOptions(),
+//                        localLogger);
+    }
+
+    private enum RawTypes {
+        MultiConf, Inv, InsDel, SegDupOrMEI, Cpx;
+    }
+
+    private static boolean hasEquallyGoodConfigurations(final AlignedContig longread) {
+        return longread.hasEquallyGoodAlnConfigurations;
+    }
+
+    private static boolean hasOnly2Alignments(final AlignedContig longreadWithOnlyOneConfig) {
+        return longreadWithOnlyOneConfig.alignmentIntervals.size() == 2;
+    }
+
+    private static boolean isSameChromosomeMapping(final AlignedContig longreadWithOnlyOneConfigAnd2Aln) {
+        return longreadWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(0).referenceSpan.getContig()
+                .equals(longreadWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(1).referenceSpan.getContig());
+    }
+
+    private static boolean isLikelyInvBreakpointOrInsInv(final AlignedContig longreadWithOnlyOneConfigAnd2AlnToSameChr) {
+        final AlignmentInterval intervalOne = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(0),
+                                intervalTwo = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(1);
+        return intervalOne.forwardStrand ^ intervalTwo.forwardStrand;
+    }
+
+    private static boolean isLikelySimpleInsDel(final AlignedContig longreadWithOnlyOneConfigAnd2AlnToSameChr) {
+        if (isLikelyInvBreakpointOrInsInv(longreadWithOnlyOneConfigAnd2AlnToSameChr)) return false;
+
+        final AlignmentInterval intervalOne, intervalTwo;
+        if (longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(0).forwardStrand) {
+            intervalOne = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(0);
+            intervalTwo = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(1);
+        } else {
+            intervalOne = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(1);
+            intervalTwo = longreadWithOnlyOneConfigAnd2AlnToSameChr.alignmentIntervals.get(0);
+        }
+        return ( intervalOne.referenceSpan.getEnd() <= intervalTwo.referenceSpan.getStart() )
+                ||
+                ( intervalOne.referenceSpan.getStart() < intervalTwo.referenceSpan.getStart() &&
+                        intervalOne.referenceSpan.getEnd() < intervalTwo.referenceSpan.getEnd());
+    }
+
+    private static boolean isLikelySegDupOrMEI(final AlignedContig longreadWithOnlyOneConfigAnd2AlnToSameChr) {
```

Comment:
SegDup (I'd call it a dispersed duplication actually) or MEI is probably the most likely explanation for a normal germline sample, but there could in rare cases be actual translocations of sequence in the same chromosome. Also, this only catches one side of a purported MEI or dispersed duplication -- a mapping that happened to be on the other side would look more like a deletion, right? A more agnostic name for these alignment patterns might be better.